docs/guidance/profiling.md

```diff
@@ -0,0 +1,65 @@
+# Profiling with KubeRay
+
+[py-spy](https://github.com/benfred/py-spy/tree/master) is a sampling profiler for Python programs. It lets you visualize what your Python program is spending time on without restarting the program or modifying the code in any way.
```

Comment:
```suggestion
## Stack trace and CPU profiling
[py-spy](https://github.com/benfred/py-spy/tree/master) is a sampling profiler for Python programs. It lets you visualize what your Python program is spending time on without restarting the program or modifying the code in any way.
```

Can we add a h2 title here so that:
- make it clear that profiling =! pyspy
- we can add more profiling-related info down the road than just py-spy, such as memory profiling, gpu profiling, etc.